src/main/java/com/commercetools/sync/services/impl/ProductServiceImpl.java

```diff
@@ -157,7 +152,7 @@ public ProductServiceImpl(@Nonnull final ProductSyncOptions syncOptions) {
     @Nonnull
     @Override
     public CompletionStage<Optional<Product>> createProduct(@Nonnull final ProductDraft productDraft) {
-        return syncOptions.applyCallbackAndCreate(productDraft, ProductCreateCommand::of, this::handleProductCreation);
+        return applyCallbackAndCreate(productDraft, ProductCreateCommand::of, this::handleProductCreation);
```

Comment:
Did you notice that now `handleProductCreation()` implementation could be moved from `ProductServiceImpl` and `CategoryServiceImpl` to base `BaseService` class?
And then you even don't need to pas this function in `applyCallbackAndCreate`, because it will be available from `BaseService`, where `applyCallbackAndCreate` is actually defined too.

Reply:
It wasn't that easy to do because the key field is not on any class that extends Resource. 
But: b79590f
unit tests to cover after refactoring: 4888c5f

Reply:
https://github.com/commercetools/commercetools-sync-java/commit/b79590ffb17c5cfde11acaa6ba590cb1f3237fe5#commitcomment-25982659